catalog_tools/analysis/estimate_beta.py

```diff
@@ -64,18 +98,47 @@ def estimate_beta_elst(magnitudes: np.ndarray) -> float:
     Args:
         magnitudes: vector of magnitudes differences, sorted in time (first
                     entry is the earliest earthquake)
+        delta_m:    discretization of magnitudes. default is no discretization
 
     Returns:
         beta:       maximum likelihood beta (b_value = beta * log10(e))
     """
-    temp_mags1 = np.append([0], magnitudes)
-    temp_mags2 = np.append(magnitudes, [0])
-    mag_diffs = temp_mags1 - temp_mags2
-    mag_diffs = mag_diffs[1:-1]
-
+    mag_diffs = differences_successive(magnitudes)
     # only take the values where the next earthquake is larger
     mag_diffs = abs(mag_diffs[mag_diffs < 0])
```

Comment:
here is the reason why you should use M2 - M1. then, this would turn into mag_diffs[mag_diffs>0], which is why the whole thing is called b POSITIVE ;)